tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -1231,6 +1231,70 @@ def get_kurtosis(y):
     return zip(index, res)
 
 
+@set_property("fctype", "combiner")
+@set_property("input", "pd.Series")
+@set_property("index_type", pd.DatetimeIndex)
+def energy_content_frequency_brackets(
+    signal: pd.Series, params: dict = {"number_of_bins": 10, "with_normalization": True}
+) -> list[tuple[str, float]]:
+    """
+
+    Calculates the FFT of the signal and groups the content of the FFT into n bins, \
+        to get an estimate of energy content per frequency bucket
+
+    Args:
+        signal (pd.Series): The distribution for which the frequency energy buckets \
+            should be calculated
+        params (_type_, optional): Parameters used for calculation of the frequency \
+            energy buckets. Parameters to be set are number_of_bins for the amount \
+                of bins, and if the values should be normalized (divided by the size \
+                    of the frequency bucket) or not.\
+            Defaults to {"number_of_bins": 10, "with_normalization": True}.
+
+    Returns:
+        list[tuple[str, float]]: _description_
+    """
+    # pylint:disable=dangerous-default-value
+
+    time_vec: np.ndarray = signal.index.to_numpy()
+    number_of_bins: int = 10
+    if "number_of_bins" in params.keys():
+        number_of_bins: int = (
+            params["number_of_bins"]
+            if (
+                params["number_of_bins"] <= len(time_vec)
+                and params["number_of_bins"] >= 0
+            )
+            else len(time_vec) - 1
+        )
+    with_normalization: bool = True
+    if "with_normalization" in params.keys():
+        with_normalization = params["with_normalization"]
+
+    freq_vec: np.ndarray = np.fft.rfftfreq(n=len(signal), d=(time_vec[1] - time_vec[0]))
+    fft_vec: np.ndarray = np.abs(np.fft.rfft(signal.to_numpy()))
+    split_freq_vec: list[np.ndarray] = np.array_split(freq_vec, number_of_bins)
+    frequency_energy_vec: list[float] = []
+    frequency_range_vec: list[tuple[float, float]] = []
+    for _, freq_entry in enumerate(split_freq_vec):
+        lower_frequency_index: int = np.where(freq_vec == freq_entry[0])[0][0]
+        upper_frequency_index: int = np.where(freq_vec == freq_entry[-1])[0][0]
+        frequency_energy_vec.append(
+            np.sum(fft_vec[lower_frequency_index:upper_frequency_index])
+            / (len(freq_entry) if with_normalization else 1.0)
+        )
+        frequency_range_vec.append((freq_entry[0], freq_entry[-1]))
+
+    return_val: list[tuple[str, float]] = [
+        (
+            f"bin_{entry_num}_max_{number_of_bins}_low_{int(frequency_range_vec[entry_num][0])}_high_{int(frequency_range_vec[entry_num][1])}",
+            entry,
+        )
+        for entry_num, entry in enumerate(frequency_energy_vec)
```

Comment:
You can use `zip` to iterate through `frequency_energy_vec` and `frequency_range_vec`at the same time